src/credentials/examples/DeviceAttestationVerifierExample.cpp

```diff
@@ -271,8 +291,22 @@ AttestationVerificationResult ExampleDACVerifier::VerifyAttestationInformation(c
     VerifyOrReturnError(attestationNonceSpan.data_equal(attestationNonce),
                         AttestationVerificationResult::kAttestationNonceMismatch);
 
+    AttestationVerificationResult error;
     ByteSpan certificationDeclarationPayload;
-    return ValidateCertificationDeclarationSignature(certificationDeclarationSpan, certificationDeclarationPayload);
+    error = ValidateCertificationDeclarationSignature(certificationDeclarationSpan, certificationDeclarationPayload);
+    VerifyOrReturnError(error == AttestationVerificationResult::kSuccess, error);
+    // TODO: Retrieve these IDs from Basic Information Cluster
+    uint16_t clusterVendorId  = 0xFFF1;
+    uint16_t clusterProductId = 0x8000;
+
+    uint16_t dacProductId = 0xFFFFu; // not specified
+    uint16_t paiProductId = 0xFFFFu; // not specified
+    VerifyOrReturnError(ExtractPIDFromX509Cert(dacCertDerBuffer, dacProductId) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kDacFormatInvalid);
+    VerifyOrReturnError(ExtractPIDFromX509Cert(paiCertDerBuffer, paiProductId) == CHIP_NO_ERROR,
+                        AttestationVerificationResult::kPaiFormatInvalid);
```

Comment:
PID is not mandatory in PAI cert.

Reply:
I am now realizing that absence is conveyed not by CHIP_ERROR_NOT_FOUND but by the value of paiProductId being a special "not found" meta-value. It's not obvious how this could fail. Suggest a comment above that expresses that missing PID will have a special value

Reply:
Addressed in d0f7bb1